### PR TITLE
Video

### PR DESCRIPTION
**Pull request by ksheremet**

### CODE REVIEW

File: app/src/main/java/ch/sheremet/dasfoo/rover/android/client/property/PropertyReader.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import android.util.Log;
+
+import java.io.InputStream;
+import java.util.Properties;
+
+/**
+ * Created by Katarina Sheremet on 6/10/16 2:48 PM.
+ */
+
+/** This class is created for temporary purpose. It will be deleted
```
Comment (dotdoom) [inline app/src/main/java/ch/sheremet/dasfoo/rover/android/client/property/PropertyReader.java:15]: Please add a `TODO` for removing this class, makes it easier to build a list of things to take care of.

File: app/src/main/java/ch/sheremet/dasfoo/rover/android/client/video/VideoDecoderRunnable.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+    private void initServerConnection() {
+        try {
+            //TODO:UriBuilder
+            URL url = new URL("https://" + mHost + ":" + mPort);
+            mUrlConnection = (HttpURLConnection) url.openConnection();
+            mUrlConnection.setRequestProperty("X-Capture-Server-PASSWORD",
+                    mPassword);
+            // Todo: change width and height
+            mUrlConnection.setRequestProperty("X-Capture-Server_WIDTH", "320");
```
Comment (dotdoom) [inline app/src/main/java/ch/sheremet/dasfoo/rover/android/client/video/VideoDecoderRunnable.java]: The format is `X-Capture-Server-WIDTH`, with dash - see `PASSWORD` above.

File: app/src/main/java/ch/sheremet/dasfoo/rover/android/client/video/VideoDecoderRunnable.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+            mUrlConnection.setRequestProperty("X-Capture-Server_HEIGHT", "240");
```
Comment (dotdoom) [inline app/src/main/java/ch/sheremet/dasfoo/rover/android/client/video/VideoDecoderRunnable.java]: Ditto.

File: app/src/main/java/ch/sheremet/dasfoo/rover/android/client/video/VideoDecoderRunnable.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    }
+
+    private void initServerConnection() {
+        try {
+            //TODO:UriBuilder
+            URL url = new URL("https://" + mHost + ":" + mPort);
+            mUrlConnection = (HttpURLConnection) url.openConnection();
+            mUrlConnection.setRequestProperty("X-Capture-Server-PASSWORD",
+                    mPassword);
+            // Todo: change width and height
```
Comment (dotdoom) [inline app/src/main/java/ch/sheremet/dasfoo/rover/android/client/video/VideoDecoderRunnable.java]: Let's stick to a single format of `TODO`, so it is easy to find action items from the code.
My suggestion is `// TODO(username): blah-blah` e.g. `// TODO(ksheremet): change width and height`, WDYT?

File: app/src/main/java/ch/sheremet/dasfoo/rover/android/client/video/MediaCodecHandler.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+                                               final int inputBufferId) {
+                ByteBuffer inputBuffer = codec.getInputBuffer(inputBufferId);
+                try {
+                    // Get Nal unit from queue
+                    byte[] unit = VideoFragment.nalQueue.take();
+                    if (unit != null) {
+                        inputBuffer.put(unit);
+                        codec.queueInputBuffer(inputBufferId, 0, unit.length, 0, 0);
+                    } else {
+                        throw new NullPointerException("Null Nal unit is gotten");
```
Comment (dotdoom) [inline app/src/main/java/ch/sheremet/dasfoo/rover/android/client/video/MediaCodecHandler.java]: As Codacy suggests, `IllegalArgumentException` could be more reasonable here.

**Comment by dotdoom** [comment]
LGTM

[Approved with PullApprove](https://pullapprove.com/dasfoo/rover-android-client/pull-request/32/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=dasfoo/rover-android-client)

**Comment by ksheremet** [comment]
LGTM

[Approved with PullApprove](https://pullapprove.com/dasfoo/rover-android-client/pull-request/32/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=dasfoo/rover-android-client)

**Pull request merged**